lib/browser/api/ipc-main.ts

```diff
@@ -1,8 +1,44 @@
 import { EventEmitter } from 'events'
+import { IpcMainEvent } from 'electron'
 
-const emitter = new EventEmitter()
+class IpcMain extends EventEmitter {
+  _invokeHandlers: {[channel: string]: (e: IpcMainEvent, ...args: any[]) => void};
+  constructor () {
+    super()
+    this._invokeHandlers = {}
+  }
+
+  handle (method: string, fn: (e: IpcMainEvent, ...args: any[]) => any) {
+    if (method in this._invokeHandlers) {
```

Comment:
This will prevent people using methods that are also the name of things on the `Object` prototype.  `_invokeHelpers` should probably become a `Map` then you can do `_invokeHelpers.has(method)`